contracts/test/PoolBurnTest.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {Currency, CurrencyLibrary} from "../libraries/CurrencyLibrary.sol";
+import {IERC20Minimal} from "../interfaces/external/IERC20Minimal.sol";
+
+import {ILockCallback} from "../interfaces/callback/ILockCallback.sol";
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+
+contract PoolBurnTest is ILockCallback {
+    using CurrencyLibrary for Currency;
+
+    IPoolManager public immutable manager;
+
+    constructor(IPoolManager _manager) {
+        manager = _manager;
+    }
+
+    // struct CallbackData
+    struct CallbackData {
+        address sender;
+        bool isMint;
+        IPoolManager.PoolKey key;
+        uint256 amount0;
+        uint256 amount1;
+    }
+
+    function mint(IPoolManager.PoolKey memory key, uint256 amount0, uint256 amount1) external {
+        manager.lock(abi.encode(CallbackData(msg.sender, true, key, amount0, amount1)));
+    }
+
+    function burn(IPoolManager.PoolKey memory key, uint256 amount0, uint256 amount1) external {
+        manager.lock(abi.encode(CallbackData(msg.sender, false, key, amount0, amount1)));
+    }
+
+    function lockAcquired(uint256, bytes calldata rawData) external returns (bytes memory) {
+        require(msg.sender == address(manager));
+
+        CallbackData memory data = abi.decode(rawData, (CallbackData));
+
+        if (data.isMint) {
+            if (data.amount0 > 0) {
+                uint256 currencyBalBefore = IERC20Minimal(Currency.unwrap(data.key.currency0)).balanceOf(data.sender);
+                uint256 managerBalBefore = manager.balanceOf(data.sender, data.key.currency0.toId());
```

Comment:
Might be more clear + consistent (with our other tests) to check the balance of the 1155 in the forge test file like this [line](https://github.com/Uniswap/v4-core/blob/571c7941914b1fb05521e31300a0068a4612358b/test/foundry-tests/PoolManager.t.sol#L895)